### ISSUE TITLE
Parameters used for the PauliNet paper

### ISSUE THREAD

**Issue by n-gao** [deepqmc/deepqmc#25]
I am trying to reproduce the results of the PauliNet paper and thought the default parameters would cover that. But when I ran 
```python
from deepqmc import Molecule, evaluate, train
from deepqmc.wf import PauliNet

mol = Molecule.from_name('Be')
net = PauliNet.from_hf(mol).cuda()
train(net)
evaluate(net)
```
I converged on `-14.661253990810394` as energy instead of the `-14.6674` from the paper for `Be`. Is there an easy way to specify the parameters of the paper?

Thank you!

**Comment by jhrmnn** [comment]
This runs the single-determinant ansatz. What you need is a multi-determinant ansatz, which can be created by passing in the specification of the [complete active space](https://en.wikipedia.org/wiki/Complete_active_space) (CAS). We've used the minimal chemically meaningful CAS, which for Be atom would be 2 electrons in 4 orbitals (the valence electrons and orbitals), so
```python
PauliNet.from_hf(mol, cas=(4, 2))
```
This argument gets actually passed to PySCF's [`CASSCF`](https://sunqm.github.io/pyscf/mcscf.html) instance.

**Comment by n-gao** [comment]
Based on my intuition about deep neural network, increasing the number of parameters should improve the results. I tried to scale up most parameters by a factor of 2, kernel_dim, number of hidden layers, etc. But the performance got slightly worse (higher variance) or the model crashed due to NaNs.
Do you have any intuitions on why this happens?

**Comment by jhrmnn** [comment]
Before moving on to the next topic, were you able to reproduce the results from the paper after passing in the appropriate CAS argument?

**Comment by n-gao** [comment]
I only tried reproducing the results on Be which I could reproduce with the cas parameter.

**Comment by jhrmnn** [comment]
That's what I meant, perfect! Thanks.

As for the scaling—you're right in finding that scaling the network parameters doesn't really help. We tried that too :) Our current understanding is that further improvements would need to come from changing the architecture more significantly, using different radial basis functions, improving the optimization scheme, or perhaps changing how exactly is the backflow is integrated into the ansatz.

**Comment by n-gao** [comment]
Thank you for the answer!
To what extend did you try scaling the parameters? For models with ~10 times the number of parameters I noticed the network to fail with the increasing part of the learning rate cycle, did removing the lr cycling help? 
Also, from quick testing with a model with ~3-4 times as many parameters, I noticed that the variance of the local energy tended to be lower, though, huge spikes occurred. In the figure below, the grey line is the 'normal' model, the orange and blue ones are trained with more parameters. 
![image](https://user-images.githubusercontent.com/9084670/89786310-03f24e00-db1c-11ea-9b2a-175a66ee8e82.png)

Do you have any intuition where these spikes come from?

Again, thank you for these discussions! They are very insightful.

**Comment by jhrmnn** [comment]
Thanks for all this interest! The Github Issues here are meant really just for technical issues with the existing code, discussions about the technical side of potential extensions, and perhaps questions about how to use the code. If you want to discuss scientific issues about new developments of the method, then we can perhaps do it as part of a collaboration, in which case please contact me or @franknoe by email.

As for the scaling, we're currently looking into it, so I can't really give any conclusions yet.
